hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/logaggregation/AggregatedLogDeletionService.java

```diff
@@ -295,14 +314,15 @@ protected Configuration createConf() {
   // as @Idempotent, it will automatically take care of RM restart/failover.
   @VisibleForTesting
   protected ApplicationClientProtocol createRMClient() throws IOException {
-    return ClientRMProxy.createRMProxy(getConfig(),
-      ApplicationClientProtocol.class);
+    return ClientRMProxy.createRMProxy(getConfig(), ApplicationClientProtocol.class);
   }
 
   @VisibleForTesting
   protected void stopRMClient() {
-    if (task != null && task.getRMClient() != null) {
-      RPC.stopProxy(task.getRMClient());
+    for (LogDeletionTask task : tasks) {
+      if (task != null && task.getRMClient() != null) {
+        RPC.stopProxy(task.getRMClient());
+      }
```

Comment:
There is only one common RMClient is created, however, you stop this RMClient three times, which could be problematic, or at best redundant. I presume the client itself is thread safe, thus safe to share between tasks, but it should only be stopped once.

Reply:
Thanks @9uapaw ,
Nice catch. 
Added a safety net for this: The tests are now verifying that the RM client is only closed once.
Please check the latest commit.